### PR TITLE
New AWS module for managing RDS instance using newer boto3 API

### PR DESCRIPTION
**Pull request by naslanidis**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

rds_v2
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel e396d5d508) last updated 2016/08/24 12:13:10 (GMT +1000)
  lib/ansible/modules/core: (detached HEAD 368ca738fa) last updated 2016/08/24 12:13:21 (GMT +1000)
  lib/ansible/modules/extras: (detached HEAD 0749ce6faa) last updated 2016/08/24 12:13:30 (GMT +1000)
```
##### SUMMARY

This module manages the creation, modification, deletion, snapshot, restore, reboot, replicate and promote of RDS Instances in AWS.

Note: Technically this is a rewrite of the existing core RDS module and while I believe rewrites using older API's aren't being included until 2.3 I thought people could still use this in the meantime, especially if support for Aurora is required. 

```
TASK [print new RDS instance] **************************************************
ok: [localhost] => {
    "msg": {
        "changed": true, 
        "rds": {
            "DBInstances": [
                {
                    "AllocatedStorage": 10, 
                    "AutoMinorVersionUpgrade": true, 
                    "AvailabilityZone": "ap-southeast-2a", 
                    "BackupRetentionPeriod": 10, 
                    "CACertificateIdentifier": "rds-ca-2015", 
                    "CopyTagsToSnapshot": false, 
                    "DBInstanceClass": "db.t2.medium", 
                    "DBInstanceIdentifier": "test-instance", 
                    "DBInstanceStatus": "available", 
                    "DBParameterGroups": [
                        {
                            "DBParameterGroupName": "test-param-group", 
                            "ParameterApplyStatus": "in-sync"
                        }
                    ], 
                    "DBSecurityGroups": [], 
                    "DBSubnetGroup": {
                        "DBSubnetGroupDescription": "test subnet group", 
                        "DBSubnetGroupName": "test-subnet-group", 
                        "SubnetGroupStatus": "Complete", 
                        "Subnets": [
                            {
                                "SubnetAvailabilityZone": {
                                    "Name": "ap-southeast-2a"
                                }, 
                                "SubnetIdentifier": "subnet-75ed1123", 
                                "SubnetStatus": "Active"
                            }, 
                            {
                                "SubnetAvailabilityZone": {
                                    "Name": "ap-southeast-2b"
                                }, 
                                "SubnetIdentifier": "subnet-6ae33123", 
                                "SubnetStatus": "Active"
                            }
                        ], 
                        "VpcId": "vpc-aad24acf"
                    }, 
                    "DbInstancePort": 0, 
                    "DbiResourceId": "db-7ZINBVLFLHNOT5ROFK76T4XXXX", 
                    "Endpoint": {
                        "Address": "test-instance2.c4nb0jta2bug.ap-southeast-2.rds.amazonaws.com", 
                        "HostedZoneId": "Z32T0VRHXEXXXX", 
                        "Port": 3306
                    }, 
                    "Engine": "mysql", 
                    "EngineVersion": "5.6.27", 
                    "InstanceCreateTime": "2016-09-03T07:36:28.832000+00:00", 
                    "LatestRestorableTime": "2016-09-03T07:40:00+00:00", 
                    "LicenseModel": "general-public-license", 
                    "MasterUsername": "testuser", 
                    "MonitoringInterval": 0, 
                    "MultiAZ": false, 
                    "OptionGroupMemberships": [
                        {
                            "OptionGroupName": "test-option-group", 
                            "Status": "in-sync"
                        }
                    ], 
                    "PendingModifiedValues": {}, 
                    "PreferredBackupWindow": "17:54-18:24", 
                    "PreferredMaintenanceWindow": "fri:12:54-fri:13:24", 
                    "PubliclyAccessible": false, 
                    "ReadReplicaDBInstanceIdentifiers": [], 
                    "StorageEncrypted": false, 
                    "StorageType": "standard", 
                    "VpcSecurityGroups": [
                        {
                            "Status": "active", 
                            "VpcSecurityGroupId": "sg-d188c123"
                        }
                    ]
                }
            ], 
            "ResponseMetadata": {
                "HTTPStatusCode": 200, 
                "RequestId": "022114a4-71aa-11e6-8a7d-bd9007adb1d5"
            }
        }
    }
}
```

### CODE REVIEW

**Comment by naslanidis** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/2845

**Comment by willthames** [comment]
duplicate_of #19523

File: lib/ansible/modules/cloud/amazon/rds_v2.py [hunk 1]
```diff
@@ -0,0 +1,1148 @@
...
+        engine=dict(type='str', choices=['mariadb', 'MySQL', 'oracle-se1', 'oracle-se', 'oracle-ee', 'sqlserver-ee', 'sqlserver-se', 'sqlserver-ex', 'sqlserver-web', 'postgres', 'aurora'], required=False),
+        master_username=dict(type='str', required=False),
+        master_user_password=dict(type='str', no_log=True, required=False),
+        vpc_security_group_ids=dict(type='list', required=False),
+        apply_immediately=dict(type='bool', default=False),
+        availability_zone=dict(type='str', required=False),
+        db_subnet_group_name=dict(type='str', required=False),
+        preferred_maintenance_window=dict(type='str', required=False),
+        db_parameter_group_name=dict(type='str', required=False),
+        backup_retention_period=dict(type=int, required=False),
```
Comment (aioue) [inline lib/ansible/modules/cloud/amazon/rds_v2.py:1087]: `int` needs singles quotes, like the others, then this works great.

**Comment by mattclay** [comment]
CI failure due to:

```
2017-02-02 00:41:00 iteritems has been removed in python3.  Alternatives:
2017-02-02 00:41:00     for KEY, VALUE in DICT.items():
2017-02-02 00:41:00     from ansible.module_utils.six import iteritems ; for KEY, VALUE in iteritems(DICT):
2017-02-02 00:41:00     from ansible.compat.six import iteritems ; for KEY, VALUE in iteritems(DICT):
2017-02-02 00:41:00 ./lib/ansible/modules/cloud/amazon/rds_v2.py:    for (k, v) in camel_params.iteritems():
2017-02-02 00:41:00 ./lib/ansible/modules/cloud/amazon/rds_v2.py:        for tag, value in module.params.get('tags').iteritems():
```

**Comment by willthames** [comment]
I would like to propose that this PR is closed, in favour of a community focus on fixing the issues with #19772 - I've spoken to @naslanidis about this and I think we're in agreement on that as the best course of action. That doesn't stop anyone using rds_v2 in their own codebase in the interim, I'd just like to put in the RDS modules that we'd have if we were basing them on current Ansible standards.

**Comment by wimnat** [comment]
based on above comment can we close_me ?

**Pull request closed without merging**